server/src/main/java/io/crate/expression/scalar/arithmetic/IntervalArithmeticFunctions.java

```diff
@@ -0,0 +1,132 @@
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ),
+            (signature, boundSignature) ->
+                new IntervalIntervalArithmeticScalar(Period::minus, signature, boundSignature)
+        );
+        module.register(
+                Signature.scalar(
+                ArithmeticFunctions.Names.MULTIPLY,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ),
+            MultiplyIntervalByIntegerScalar::new
+        );
+        module.register(
+                Signature.scalar(
+                ArithmeticFunctions.Names.MULTIPLY,
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTERVAL.getTypeSignature()
+            ),
+            MultiplyIntervalByIntegerScalar::new
+        );
+    }
+
+    private static class IntervalIntervalArithmeticScalar extends Scalar<Period, Object> {
+
+        private final BiFunction<Period, Period, Period> operation;
+
+        IntervalIntervalArithmeticScalar(BiFunction<Period, Period, Period> operation,
+                                         Signature signature,
+                                         BoundSignature boundSignature) {
+            super(signature, boundSignature);
+            this.operation = operation;
+        }
+
+        @Override
+        public Period evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+            Period fst = (Period) args[0].value();
+            Period snd = (Period) args[1].value();
+
+            if (fst == null || snd == null) {
+                return null;
+            }
+            return operation.apply(fst, snd);
+        }
+    }
+
+    private static class MultiplyIntervalByIntegerScalar extends Scalar<Period, Object> {
+
+        private final int integerIdx;
+        private final int periodIdx;
+
+        MultiplyIntervalByIntegerScalar(Signature signature, BoundSignature boundSignature) {
+            super(signature, boundSignature);
+            var firstArgType = boundSignature.argTypes().get(0);
+            if (firstArgType.id() == IntervalType.ID) {
+                periodIdx = 0;
+                integerIdx = 1;
+            } else {
+                periodIdx = 1;
+                integerIdx = 0;
+            }
+        }
+
+        @Override
+        public Period evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+            Integer integer = (Integer) args[integerIdx].value();
+            Period period = (Period) args[periodIdx].value();
+
+            if (integer == null || period == null) {
+                return null;
+            }
+            return period.multipliedBy(integer);
```

Comment:
Hi @piotrrzysko, thanks a lot for your contribution!

Regarding the example you mentioned in the PR description:

```
SELECT INTERVAL '1 second' * 900;
returns:

0 years 0 mons 0 days 0 hours 0 mins 900.0 secs
instead of:

0 years 0 mons 0 days 0 hours 15 mins 0.0 secs

```

PG 14 returns 15 mins in this case.

Could you please try out  `period.multipliedBy(integer).normalizedStandard(PeriodType.yearMonthDayTime())`?

This would be aligned with out timestamp subtraction/interval average result normalization and hopefully will slightly improve normalisation on multiplication. Maybe we can add such scenario to the tests.



On overflow: I agree that we can address this as a follow up.